xwiki-platform-core/xwiki-platform-resource/xwiki-platform-resource-default/src/main/java/org/xwiki/resource/internal/AbstractResourceReferenceHandlerManager.java

```diff
@@ -88,10 +88,17 @@ public void handle(ResourceReference reference) throws ResourceReferenceHandlerE
 
     @Override
     public boolean canHandle(T resourceReferenceQualifier)
+    {
+        // FIXME: this is not good, it means we don't support backward compatibility.
```

Comment:
Reviewers: that's the part I'm not a big fan of. I see two choices to fix it:
  1. we have two branches in the code, one using the full `ResourceReference` to check the handler (which is the current implem you can see below), and another one using the old implem which only uses the type `T` by extracting the ResourceReferenceQualifier. It probably means a bit of duplicating code, for something which I'm not even sure is implemented somewhere (we're talking about an implem of `ResourceReferenceHandlerManager` here, not just an handler)
  2. We break the API of `ResourceReferenceHandlerManager` to remove entirely `#canHandle(T resourceReferenceQualifier)`, maybe we'll break someone (I doubt it), but at least it's clear what we're doing and we don't need to maintain two artificial branches.

Reply:
When strictly speaking about retro compatibility another thing to worry about would not so much be the implementation of #canHandle(T) (since this method would never been called anymore by XWiki Standard) but the default implementation of the new `ResourceReferenceHandlerManager#canHandle(ResourceReference)` which is close to useless right now since it will never have the expected behavior for a ResourceReferenceHandlerManager provided by some extension since this extension will only implement the old one.

That being said, I'm pretty sure the chance that anyone implemented or is planning to implement `ResourceReferenceHandlerManager` in an extension is very close to 0.

Back to your question: the chance that anyone directly use ResourceReferenceHandlerManager at all in an extension is most probably very close to 0 too, so I'm fine with `2.`